src/vm/node_modules/VM.js

```diff
@@ -6342,6 +6342,11 @@ function buildNicZonecfg(vmobj, payload, log)
                 + nic.allowed_dhcp_cids.join(',') + '")\n';
         }
 
+        if (nic.hasOwnProperty('pci_slot')) {
```

Comment:
How are we accountin for the `primary` property? Should that nic not be kept in the first slot as it currently is?

If i remember the boot.c for bhyve currently the primary nic is always slot 0. (I have not seen your boot.c changes as I can't find the PR for that atm?)

Reply:
It should still work the same way as today -- we aren't requiring `pci_slot` for NICs, so if omitted, the NIC goes in the next available function in bus 0, slot 6